server/src/main/java/keywhiz/ServiceModule.java

```diff
@@ -82,32 +86,6 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     return new SimpleLogger();
   }
 
-  // ManagedDataSource
-
-  @Provides @Singleton ManagedDataSource dataSource(Environment environment,
-      KeywhizConfig config) {
-    DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
-    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "db-writable");
-    environment.lifecycle().manage(dataSource);
-
-    environment.healthChecks().register("db-read-write-health",
-        new JooqHealthCheck(dataSource, LOG_ONLY));
-
-    return dataSource;
-  }
-
-  @Provides @Singleton @Readonly ManagedDataSource readonlyDataSource(Environment environment,
-      KeywhizConfig config) {
-    DataSourceFactory dataSourceFactory = config.getReadonlyDataSourceFactory();
-    ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "db-readonly");
-    environment.lifecycle().manage(dataSource);
-
-    environment.healthChecks().register("db-readonly-health",
-        new JooqHealthCheck(dataSource, RETURN_UNHEALTHY));
-
-    return dataSource;
-  }
```

Comment:
The only thing that differs from the main service (DropWizard's built-in "server" command) and our hand-rolled commands is that they define their data sources differently (different names, and only the service data sources have health checks). In order to seamlessly support both use cases we extract the service-specific data source bindings into a dedicated module so that they're only bound on the service code path.